Improve loading sequence for the theme

awinny opened the pull request:
**Describe pull-request**
- Updated sass dependencies, causing issues when using newer version of node
- Added a display check for elements before showing  them

</br>**Solving issue**
Add which issue this pull-request solves by adding # plus the number of the issue (for example #123)
Fixes: #184

awinny committed:
Improve loading sequence for the theme
- Added a extra check before rendering the component template

awinny committed:
Merge branch 'master' into improvement/Improve_loading_sequence_for_the_theme

awinny committed:
Changed fouc solution from javascript to pure css
- Updated sass dependencies, causing issues when using newer version of node
- Reverted javascript solution for hiding/showing a component on load in favor of a pure css setup
- Refactoring

helloimela commented:
Should we add the `--display:none` in c-theme `:root { }` for initial display value?

awinny committed:
Added default display type
- Added missing default display type to the theme component

awinny committed:
Removed old defaults
- Removed old default display type from components
- Added condition for when display none should be set initally for components

helloimela commented:
It does not work in IE, somehow, the generation of styles in IE is different that the one we used to have. Is it because upgrading stencil-sass ?

![image](https://user-images.githubusercontent.com/1199101/66570019-020ed580-eb6d-11e9-9c87-8e9480156e6a.png)

helloimela commented on src/components.scss:
```diff
@@ -1,5 +1,5 @@
 :host {
-  display: block;
+  display: var(--display);
```
It seems like the problem with IE caused by this line. Somehow when we have css vars in global component scss, it will render unique class name for IE styling (such as sc-c-header-0).

helloimela reviewed (changes requested):
Fix problems in IE and fix conflicts. Probably wants to wait merge this after Stencil One task is done.

awinny commented:
This was merged in to the branch feature/Stencil_upgrade
